oak/client/sgx_application_client.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2019 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef OAK_CLIENT_SGX_APPLICATION_CLIENT_H_
+#define OAK_CLIENT_SGX_APPLICATION_CLIENT_H_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "asylo/grpc/auth/enclave_credentials_options.h"
+#include "asylo/identity/enclave_assertion_authority_configs.h"
+#include "include/grpcpp/grpcpp.h"
+
+namespace oak {
+
+class SgxApplicationClient {
+ public:
+  SgxApplicationClient(
```

Comment:
Document what `mrenclave_strings` is used for, i.e. that it verifies the remote attestation and then only accepts an attestation in which MRENCLAVE is one of the values provided here.